ui/StatusQ/src/StatusQ/Popups/StatusMenuItemDelegate.qml

```diff
@@ -7,187 +7,164 @@ import StatusQ.Components 0.1
-        anchors.leftMargin: 4
+        readonly property real arrowPadding: root.spacing + (root.subMenu && root.arrow ? root.arrow.width : 0)
+        readonly property real indicatorPadding: root.spacing + (root.indicator.visible ? root.indicator.width : 0)
+
+        leftPadding: !root.mirrored ? indicatorPadding : arrowPadding
+        rightPadding: root.mirrored ? indicatorPadding : arrowPadding
 
         horizontalAlignment: Text.AlignLeft
         verticalAlignment: Text.AlignVCenter
 
-        text: statusPopupMenuItem.text
-        color: {
-            switch (statusPopupMenuItem.action.type) {
-                case StatusMenuItem.Type.Danger:
-                  return Theme.palette.dangerColor1
-                default:
-                  return Theme.palette.directColor1
-            }
-        }
-        font.pixelSize: !!statusPopupMenuItem.action.fontSettings ? statusPopupMenuItem.action.fontSettings.pixelSize : 13
-        font.bold: !!statusPopupMenuItem.action.fontSettings ? statusPopupMenuItem.action.fontSettings.bold : false
-        font.italic: !!statusPopupMenuItem.action.fontSettings ? statusPopupMenuItem.action.fontSettings.italic : false
+        text: root.text
+        color: !root.enabled ? Theme.palette.baseColor1
+                             : d.isDangerIcon ? Theme.palette.dangerColor1 : Theme.palette.directColor1
+
+        font.pixelSize: d.fontSettings.pixelSize
+        font.bold: d.fontSettings.bold
+        font.italic: d.fontSettings.italic
         elide: Text.ElideRight
-        visible: statusPopupMenuItem.action.enabled
+        visible: true
     }
 
     arrow: StatusIcon {
         id: arrowIcon
-        anchors.verticalCenter: parent.verticalCenter
-        anchors.right: parent.right
-        anchors.rightMargin: 8
+        x: root.mirrored ? root.leftPadding : root.width - width - root.rightPadding
+        y: root.topPadding + (root.availableHeight - height) / 2
+
         height: 16
-        visible: statusPopupMenuItem.subMenu
+        visible: d.isSubMenu
         icon: "next"
         color: Theme.palette.directColor1
     }
 
     background: Rectangle {
         color: {
-            if (statusPopupMenuItem.hovered) {
-                return statusPopupMenuItem.action.type === StatusMenuItem.Type.Danger ? Theme.palette.dangerColor3 : Theme.palette.statusPopupMenu.hoverBackgroundColor
-            }
-            return "transparent"
+            if (!root.hovered && !d.subMenuOpened)
+                return "transparent"
+            if (d.isStatusAction && root.action.type === StatusMenuItem.Type.Danger)
+                return Theme.palette.dangerColor3;
+            return Theme.palette.statusPopupMenu.hoverBackgroundColor;
         }
     }
 
     MouseArea {
-        id: sensor
         anchors.fill: parent
         cursorShape: Qt.PointingHandCursor
-        hoverEnabled: statusPopupMenuItem.action.enabled
+        hoverEnabled: root.enabled
         onPressed: mouse.accepted = false
```

Comment:
I'd remove this onPressed handler; you can achieve the same with setting `acceptedButtons: Qt.NoButton`